src/cli/telemetry/helpers.ts

```diff
@@ -0,0 +1,51 @@
+import { getCompilerSystem } from '../state/stencil-cli-config';
+
+interface TerminalInfo {
+  /**
+   * Whether this is in CI or not.
+   */
+  readonly ci: boolean;
+  /**
+   * Whether the terminal is an interactive TTY or not.
+   */
+  readonly tty: boolean;
+}
+
+export declare const TERMINAL_INFO: TerminalInfo;
+
+export const tryFn = async <T extends (...args: any[]) => Promise<R>, R>(fn: T, ...args: any[]): Promise<R | null> => {
+  try {
+    return await fn(...args);
+  } catch {
+    // ignore
+  }
+
+  return null;
+};
+
+export const isInteractive = (object?: TerminalInfo): boolean => {
+  const terminalInfo =
+    object ||
+    Object.freeze({
+      tty: process.stdout.isTTY ? true : false,
+      ci:
+        ['CI', 'BUILD_ID', 'BUILD_NUMBER', 'BITBUCKET_COMMIT', 'CODEBUILD_BUILD_ARN'].filter(v => !!process.env[v])
+          .length > 0 || process.argv.includes('--ci'),
```

Comment:
`process` is a Node built-in, so you'd have to access these through the compiler system. You should be able to replace `process.env[v]` with `CompilerSystem.getEnvironmentVar(v)`

Reply:
Ahh, hmm... I'm going to need to do a couple changes here in that case.